### ISSUE TITLE
Redirect to new repo URL when visit with old reponame

### ISSUE THREAD

**Issue by Unknwon** [gogits/gogs#1566]
Currently, after renaming a repository, the dashboard feeds became useless as they always result in 404.

To solve it, update all actions' data is not acceptable since it violates the purpose and definition of `action`, which is the information of a happened event.

So, 

1. Add a redirect rule every time renamed a repository.
2. Delete corresponding redirect rule when another repository is created with same name.

**Comment by Naxiz** [comment]
I don't think this is the right way to solve this problem. This way it'll still be a mess, and it'll be confusing when a new repository will suddenly claim the name later.

The problem should be solved at the root of the cause: when renaming a repository, rename all references to it as well.

Even better: use the foreign keys (drop the `repo_user_name` and `repo_name` columns, as there's already an `repo_id` and `user_id` column), so by simply renaming the repository all references to it will automatically get updated.

The `repo_user_name` and `repo_name` columns are already redundant data, which will only cause problems if `repo_id` and `repo_name` get out of sync.
